### PR TITLE
feat: Add headers param support in CPL [DHIS2-13511]

### PR DESCRIPTION
**Pull request by maikelarabori**
This feature aims to add support for the `headers` request param.

When this param is set, the response should respect the `headers` specified, returning only the defined ones in the respective order. The main change is the code related to this class `dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/shared/GridHeaders.java`

I also did a small code clean-up as part of this PR.

_You will notice a small commit from "mortenoh", that can be ignored. This is because I merged `master` into my commit._

### CODE REVIEW

File: dhis-2/dhis-api/src/main/java/org/hisp/dhis/security/apikey/ApiTokenAttribute.java [hunk 1]
```diff
@@ -45,13 +44,18 @@
     @JsonSubTypes.Type( value = RefererAllowedList.class, name = "RefererAllowedList" ),
     @JsonSubTypes.Type( value = MethodAllowedList.class, name = "MethodAllowedList" ) } )
 @JacksonXmlRootElement( localName = "apiTokenAttribute", namespace = DxfNamespaces.DXF_2_0 )
-public abstract class ApiTokenAttribute implements Serializable
+public abstract class ApiTokenAttribute implements EmbeddedObject
```
Comment (d-bernat) [inline dhis-2/dhis-api/src/main/java/org/hisp/dhis/security/apikey/ApiTokenAttribute.java:47]: why did you remove Serializable interface?

**Review by d-bernat — APPROVED** [review]

File: dhis-2/dhis-api/src/main/java/org/hisp/dhis/security/apikey/ApiTokenAttribute.java (hunk 1 above)
Comment (maikelarabori) [reply dhis-2/dhis-api/src/main/java/org/hisp/dhis/security/apikey/ApiTokenAttribute.java:47]: As discussed, this change is not part of this PR.

**Review by gnespolino — APPROVED** [review]
